src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -58,13 +60,37 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
         for charge in expunger.acquittals:
             charge.set_time_ineligible('Recommend sequential expungement', eligibility_date)
 
+    @staticmethod
+    def _calculate_has_subsequent_charge(class_b_felony, charges):
+        other_charges = [charge for charge in charges if charge != class_b_felony]
+        for other_charge in other_charges:
+            if other_charge.acquitted():
+                date_of_other_charge = other_charge.date
+            else:
+                date_of_other_charge = other_charge.disposition.date
+
+            if date_of_other_charge > class_b_felony.disposition.date:
+                return True
+        return False
+
     @staticmethod
     def _evaluate_class_b_felonies(expunger):
-        if expunger.most_recent_charge and expunger.most_recent_charge.disposition.date > TimeAnalyzer.TWENTY_YEARS_AGO:
-            for charge in expunger.class_b_felonies:
-                charge.set_time_ineligible('Time-ineligible under 137.225(5)(a)(A)(i)',
-                                           expunger.most_recent_charge.disposition.date + relativedelta(
-                                               years=TimeAnalyzer.TWENTY_YEARS))
+        for class_b_felony in expunger.class_b_felonies:
+            # If the class B felony is acquitted, then we have already handled its time eligibility.
+            # If it's convicted, this restriction overrides any of those rules.
+            if not class_b_felony.acquitted():
```

Comment:
It's usually recommended to use positive expressions if available. Perhaps we can say `if class_b_felony.convicted()` or if we also need to handle unknown then the expression can be if convicted or unknown.

Reply:
Sure. we don't do anything here the disposition is missing or unknown (they're not subject to time analysis), so this check can just become `convicted()`.